### ISSUE TITLE
[bug]:  Error with flutterfire_cli When Archiving App in Xcode

### ISSUE THREAD

**Issue by LeonManolo** [invertase/flutterfire_cli#321]
### Is there an existing issue for this?

- [X] I have searched the existing issues.

### CLI Version

1.0.1-dev.1 or 1.0.0 (both not working)

### Firebase Tools version

13.11.4

### Flutter Doctor Output

[✓] Flutter (Channel stable, 3.22.2, on macOS 13.6.7 22G720 darwin-x64, locale en-DE)
    • Flutter version 3.22.2 on channel stable at /Users/manolo/Developer/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 761747bfc5 (3 weeks ago), 2024-06-05 22:15:13 +0200
    • Engine revision edd8546116
    • Dart version 3.4.3
    • DevTools version 2.34.3

[!] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/manolo/Library/Android/sdk
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for more details.

[✓] Xcode - develop for iOS and macOS (Xcode 15.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15C500b
    • CocoaPods version 1.13.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2023.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-17.0.7b1000.6-10550314)

[✓] IntelliJ IDEA Ultimate Edition (version 2022.3.2)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin version 223.8888

[✓] IntelliJ IDEA Community Edition (version 2022.1.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.90.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.90.0

[✓] Connected device (3 available)
    • Leon-Manolo iPhone (mobile) • 00008101-001145A001E9001E • ios            • iOS 17.5.1 21F90
    • macOS (desktop)             • macos                     • darwin-x64     • macOS 13.6.7 22G720 darwin-x64
    • Chrome (web)                • chrome                    • web-javascript • Google Chrome 126.0.6478.116

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.

### Description

I am encountering a persistent issue with flutterfire_cli when trying to archive my Flutter application in Xcode. Despite successful builds using flutter build ios --release and the ability to run the app on my iPhone using flutter run, the archiving process fails with the following error:

```
Can't load Kernel binary: Invalid kernel binary format version.
ERR: flutterfire_cli as globally activated doesn't support Dart 2.18.5, try: dart pub global activate flutterfire_cli
```

**Attempts to Resolve the Issue:**

1. Ran dart pub global activate flutterfire_cli to update flutterfire_cli.
2. Checked that Flutter, Dart, and all related packages are up to date.
3. Performed flutter clean and flutter pub get.
4. Checked for any path discrepancies or misconfigurations.

Despite these efforts, the issue persists. Any insights or suggestions to resolve this error would be greatly appreciated.

**Environment:**

1. Flutter Version: 3.22.2
2. Dart Version: 3.4.3
3. Xcode Version:  15.2 (15C500b)
4. macOS Version: Ventura 13.6.7 (22G720)

### Steps to reproduce

1. Run flutter build ios --release.
2. Attempt to archive the app in Xcode.

### Expected behavior

The app archives successfully without errors.

### Screenshots

![Screenshot 2024-06-26 at 12 54 44](https://github.com/invertase/flutterfire_cli/assets/87019114/2ef6e09c-9c73-4929-87e3-c00de73aa6d7)

![Screenshot 2024-06-26 at 12 55 48](https://github.com/invertase/flutterfire_cli/assets/87019114/bf4b8ecb-4dbd-427d-ae35-b5986f449883)

### Additional context and comments

Happens when the custom shell script: 'FlutterFire: "flutterfire upload-crashlytics-symbols"' gets executed.

**Comment by russellwheatley** [comment]
Seems very strange, I just ran the command in a test app and it was successful:
```
➜  flutter_test_cli git:(main) ✗ flutter build ios --release
Building com.example.flutterTestCli for device (ios-release)...
Automatically signing iOS for device deployment using specified development team in Xcode project: YYX2P3XVJ7
Running Xcode build...                                                  
 └─Compiling, linking and signing...                        13.4s
Xcode build done.                                           49.4s
✓ Built build/ios/iphoneos/Runner.app (28.5MB)
```

As you can see from the pubspec, Dart 2.18.5 is supported by the FlutterFire CLI: https://github.com/invertase/flutterfire_cli/blob/main/packages/flutterfire_cli/pubspec.yaml#L7

Seems like an environment issue, are you sure you don't have another Dart SDK somewhere on your path, perhaps?

**Comment by LeonManolo** [comment]
> Seems very strange, I just ran the command in a test app and it was successful:
> 
> ```
> ➜  flutter_test_cli git:(main) ✗ flutter build ios --release
> Building com.example.flutterTestCli for device (ios-release)...
> Automatically signing iOS for device deployment using specified development team in Xcode project: YYX2P3XVJ7
> Running Xcode build...                                                  
>  └─Compiling, linking and signing...                        13.4s
> Xcode build done.                                           49.4s
> ✓ Built build/ios/iphoneos/Runner.app (28.5MB)
> ```
> 
> As you can see from the pubspec, Dart 2.18.5 is supported by the FlutterFire CLI: https://github.com/invertase/flutterfire_cli/blob/main/packages/flutterfire_cli/pubspec.yaml#L7
> 
> Seems like an environment issue, are you sure you don't have another Dart SDK somewhere on your path, perhaps?

Yes, running `flutter build ios --release` also works for me. The problem is trying to archive/build the project in Xcode.
```
Running Xcode build...                                                  
 └─Compiling, linking and signing...                        38.3s
Xcode build done.                                           278.0s
✓ Built build/ios/iphoneos/Runner.app (63.3MB)
```

I have only one version of Dart installed (Version 3.4.3). I double-checked that.
Running `dart --version` confirms that too:
```
➜  ~ dart --version
Dart SDK version: 3.4.3 (stable) (Tue Jun 4 19:51:39 2024 +0000) on "macos_x64"
```

**Comment by LeonManolo** [comment]
> Seems very strange, I just ran the command in a test app and it was successful:
> 
> ```
> ➜  flutter_test_cli git:(main) ✗ flutter build ios --release
> Building com.example.flutterTestCli for device (ios-release)...
> Automatically signing iOS for device deployment using specified development team in Xcode project: YYX2P3XVJ7
> Running Xcode build...                                                  
>  └─Compiling, linking and signing...                        13.4s
> Xcode build done.                                           49.4s
> ✓ Built build/ios/iphoneos/Runner.app (28.5MB)
> ```
> 
> As you can see from the pubspec, Dart 2.18.5 is supported by the FlutterFire CLI: https://github.com/invertase/flutterfire_cli/blob/main/packages/flutterfire_cli/pubspec.yaml#L7
> 
> Seems like an environment issue, are you sure you don't have another Dart SDK somewhere on your path, perhaps?

Running `dart --version` inside the build phase, shows that the dart version is lower than specified in the pubspec.yaml:
```yaml
environment:
  sdk: '>=3.0.0 <4.0.0'
  flutter: ">=3.7.0"
```


build phase:
![Screenshot 2024-07-02 at 12 35 17](https://github.com/invertase/flutterfire_cli/assets/87019114/186477e1-a830-49a7-9435-3c45f1aba26d)

output after changing the buildphase:
```
THIS IS THE DART VERSION:
Dart SDK version: 2.18.5 (stable) (Tue Nov 22 15:47:29 2022 +0000) on "macos_x64"
Installed flutterfire_cli version:
Can't load Kernel binary: Invalid kernel binary format version.
FINE: Pub 2.18.5
ERR : flutterfire_cli as globally activated doesn't support Dart 2.18.5, try: dart pub global activate flutterfire_cli
FINE: Exception type: DataException
FINE: package:pub/src/utils.dart 531:36                      dataError
    | package:pub/src/global_packages.dart 383:9             GlobalPackages.find.<fn>
    | dart:collection                                        _LinkedHashMapMixin.forEach
    | package:pub/src/global_packages.dart 376:29            GlobalPackages.find
...
```

Might you have any idea why that is? It crashes as soon as any `flutterfire` gets executed

**Comment by Mo-Meshrif** [comment]
Try this solution, it worked with me.
![Screenshot 2024-09-22 at 10 19 45 AM](https://github.com/user-attachments/assets/546812b3-711e-4893-acd7-7870d7bfc372)

**Comment by otaviobrudden** [comment]
I was facing the same issue, but with vscode.
Following @LeonManolo steps, I figured out that the dart version and path was different from my environment.

XCODE DART ~VERSION~ (PATH)
![telegram-cloud-photo-size-1-4916228216969080743-x](https://github.com/user-attachments/assets/771b172b-26a7-46df-9a60-a9f32af00575)

Going to that directory I found that it is a symbolic link

`/opt/homebrew/bin/dart -> ../Cellar/dart/2.18.2/bin/dart`

I searched for similar issues caused by this directory/dart and found this stackoverflow related thread about running `flutter doctor`: https://stackoverflow.com/a/76925977

My `flutter doctor` output. Look at the first warning:

```Doctor summary (to see all details, run flutter doctor -v):
[!] Flutter (Channel stable, 3.22.2, on macOS 14.2.1 23C71 darwin-arm64, locale pt-BR)
    ! Warning: dart on your path resolves to /opt/homebrew/Cellar/dart/2.18.2/libexec/bin/dart, which is
      not inside your current Flutter SDK checkout at /Users/movlabs/fvm/versions/3.22.2. Consider adding
      /Users/movlabs/fvm/versions/3.22.2/bin to the front of your path.
[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 15.4)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2022.2)
[✓] IntelliJ IDEA Community Edition (version 2021.2.4)
[✓] VS Code (version 1.94.0)
[✓] Connected device (4 available)
    ! Error: Browsing on the local area network for iPhone de Movement. Ensure the device is unlocked and
      attached with a cable or associated with the same local area network as this Mac.
      The device must be opted into Developer Mode to connect wirelessly. (code -27)
[✓] Network resources

! Doctor found issues in 1 category.```

As Lydia suggests on stackoverflow, I uninstalled dart using brew:
`brew uninstall dart`

Running `flutter doctor` again:
```Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.22.2, on macOS 14.2.1 23C71 darwin-arm64, locale pt-BR)
[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 15.4)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2022.2)
[✓] IntelliJ IDEA Community Edition (version 2021.2.4)
[✓] VS Code (version 1.94.0)
[✓] Connected device (4 available)
    ! Error: Browsing on the local area network for iPhone de Movement. Ensure the device is unlocked and attached with a cable or associated with the same local area network as this Mac.
      The device must be opted into Developer Mode to connect wirelessly. (code -27)
[✓] Network resources

• No issues found!```

After that my build worked.

Thank you, guys.

**Comment by otavioarrudasantos** [comment]
I was facing the same issue, but with vscode.

Following @LeonManolo steps, I figured out that the dart version and path was different from my environment.

using `brew uninstall dart` solved my issue.

Here are the steps I followed:

XCODE DART ~VERSION~ (PATH)
![telegram-cloud-photo-size-1-4916228216969080743-x](https://github.com/user-attachments/assets/771b172b-26a7-46df-9a60-a9f32af00575)

Going to that directory I found that it is a symbolic link

`/opt/homebrew/bin/dart -> ../Cellar/dart/2.18.2/bin/dart`

I searched for similar issues caused by this directory/dart and found this stackoverflow related thread about running `flutter doctor`: https://stackoverflow.com/a/76925977

My `flutter doctor` output. Look at the first warning:

```Doctor summary (to see all details, run flutter doctor -v):
[!] Flutter (Channel stable, 3.22.2, on macOS 14.2.1 23C71 darwin-arm64, locale pt-BR)
    ! Warning: dart on your path resolves to /opt/homebrew/Cellar/dart/2.18.2/libexec/bin/dart, which is
      not inside your current Flutter SDK checkout at /Users/movlabs/fvm/versions/3.22.2. Consider adding
      /Users/movlabs/fvm/versions/3.22.2/bin to the front of your path.
[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 15.4)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2022.2)
[✓] IntelliJ IDEA Community Edition (version 2021.2.4)
[✓] VS Code (version 1.94.0)
[✓] Connected device (4 available)
    ! Error: Browsing on the local area network for iPhone de Movement. Ensure the device is unlocked and
      attached with a cable or associated with the same local area network as this Mac.
      The device must be opted into Developer Mode to connect wirelessly. (code -27)
[✓] Network resources

! Doctor found issues in 1 category.

```


As Lydia suggests on stackoverflow, I uninstalled dart using brew:
`brew uninstall dart`

Running `flutter doctor` again:

```Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.22.2, on macOS 14.2.1 23C71 darwin-arm64, locale pt-BR)
[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 15.4)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2022.2)
[✓] IntelliJ IDEA Community Edition (version 2021.2.4)
[✓] VS Code (version 1.94.0)
[✓] Connected device (4 available)
    ! Error: Browsing on the local area network for iPhone de Movement. Ensure the device is unlocked and attached with a cable or associated with the same local area network as this Mac.
      The device must be opted into Developer Mode to connect wirelessly. (code -27)
[✓] Network resources

• No issues found!

```

After that my build worked.

Thank you, guys.
